torch/ao/pruning/sparsifier/base_structured_pruner.py

```diff
@@ -0,0 +1,60 @@
+from typing import Set, Type
+import torch
+from torch import nn
+from torch.nn.utils import parametrize
+
+from torch.ao.pruning import BaseSparsifier
+from .utils import FakeStructuredSparsity, BiasHook
+
+__all__ = ["BaseStructuredPruner"]
+
+SUPPORTED_STRUCTURED_PRUNING_MODULES = {  # added to config if None given
+    nn.Linear,
+    nn.Conv2d,
+}
+
+class BaseStructuredPruner(BaseSparsifier):
```

Comment:
we should keep the naming consistent, if the parent is `BaseSparsifier` the child should be `BaseStructuredSparsifier`.  The name change should be a separate PR.

Reply:
I think we should take a step further, and rename all to "*Pruner". The reason we had `Sparsifier` and `Pruner` was because of the outdated definitions: "sparity" was ="unstructured", while "pruning was ="structured". But now those definitions are not applicable anymore. Could be in future PRs though